src/vips/Image_methods.lua

```diff
@@ -575,6 +575,28 @@ end
 
 -- convenience functions
 
+function Image_method:hasalpha()
+    local result = vips_lib.vips_image_hasalpha(self.vimage)
+    if result == nil then
+        error(verror.get())
+    end
+
```

Comment:
```suggestion
```
[`vips_image_hasalpha`](https://www.libvips.org/API/current/VipsImage.html#vips-image-new-from-memory) can't return NULL (which would be `ffi.NULL`), so please remove this check.

Reply:
Alright, done